carp.common/src/commonMain/kotlin/dk/cachet/carp/common/DateTime.kt

```diff
@@ -17,27 +13,22 @@ expect class DateTime( msSinceUTC: Long )
 {
     val msSinceUTC: Long
 
+
     companion object
     {
         fun now(): DateTime
     }
 
+
     /**
      * Output as ISO 8601 UTC date and time in extended format with day and seconds precision with 3 decimal places following a period.
      * E.g., "2020-01-01T12:00:00.000Z"
      */
-    override fun toString(): String
```

Comment:
almost all date/libs will return the ISO string on `toString() ` . It is even lossless in the sense that the seconds fraction will provide the precision available. I think you should stick to that convention. It will not violate the (sound) principles you outline in the PR description.

Reply:
I was wondering about that! I tried it and got an equality exception which I expected was due to rounding, but maybe there was another cause.

Will revisit.

Reply:
Solved in https://github.com/cph-cachet/carp.core-kotlin/pull/195/commits/a390e8ce7763a1d22526067a784506b2948a8b93